engine/src/build/config/mac/BUILD.gn

```diff
@@ -28,3 +28,53 @@ config("mac_dynamic_flags") {
 config("mac_executable_flags") {
   ldflags = [ "-Wl,-pie" ]  # Position independent.
 }
+
+config("compiler") {
+  # swiftc -sdk flag.
+  _sdk_root = mac_sdk_path
+  if (create_xcode_symlinks) {
+    _sdk_root = rebase_path(mac_sdk_path, root_build_dir)
+  }
+
+  # swiftc -target flag (target triplet).
+  # CPU architecture.
+  if (current_cpu == "x64") {
+    _triplet_cpu = "x86_64"
+  } else if (current_cpu == "arm64") {
+    _triplet_cpu = "arm64"
+  } else {
+    # x86, arm, armv7, etc.
+    assert(false, "unsupported cpu: $current_cpu")
+  }
+  # Environment.
+  _triplet_environment = ""
+  # OS.
+  _triplet_os = "apple-macos"
+  _target_triplet = "$_triplet_cpu-$_triplet_os$mac_deployment_target$_triplet_environment"
```

Comment:
This could be a little more succinct, but this makes it easier to potentially merge this logic with the equivalent iOS logic in //build/config/iOS/BUILD.gn.